samples/ingestion/ingestion-client/FetchTranscription/Language/AnalyzeConversationsProvider.cs

```diff
@@ -219,6 +220,43 @@ public async Task<bool> ConversationalRequestsCompleted(IEnumerable<AudioFileInf
             return errors;
         }
 
+        private static IEnumerable<ErrorEntity> GetAllErrorsFromResults((IEnumerable<AnalyzeConversationPiiResults> piiResults, IEnumerable<AnalyzeConversationSummarizationResults> summarizationResults, IEnumerable<string> errors)[] results)
+        {
+            var resultErrors = new List<ErrorEntity>();
```

Comment:
since there seems to be a lot of if's here and also responses which can be null we can get nulls, it might be worth updating it to account for nullability something of this format would be more resilient.


```
private static IEnumerable<ErrorEntity> GetAllErrorsFromResults((IEnumerable<AnalyzeConversationPiiResults> piiResults, IEnumerable<AnalyzeConversationSummarizationResults> summarizationResults, IEnumerable<string> errors)[] results)
        {
            var resultErrors = new List<ErrorEntity>();

            if (results == null || !results.Any())
            {
                return resultErrors;
            }

            foreach (var (piiResults, summarizationResults, errors) in results)
            {
                var piiErrors = piiResults?
                    .Where(pr => pr.Errors?.Any() ?? false)?
                    .SelectMany(e => e.Errors);

                if (piiErrors != null)
                {
                    resultErrors.AddRange(piiErrors);
                }

                var summarizeErrors = summarizationResults?
                    .Where(sr => sr.Errors?.Any() ?? false)?
                    .SelectMany(sr => sr.Errors);

                if (summarizeErrors != null)
                {
                    resultErrors.AddRange(summarizeErrors);
                }
            }

            return resultErrors;
        }
```

Reply:
updated to the code you shared below